datacapture/src/test/java/com/google/android/fhir/datacapture/QuestionnaireViewModelParameterizedTest.kt

```
@@ -183,8 +184,23 @@ class QuestionnaireViewModelParameterizedTest(
     val printer: IParser = FhirContext.forR4().newJsonParser()
 
     fun <T : IBaseResource> assertResourceEquals(actual: T, expected: T) {
```

as @ndegwamartin pointed out, might as well split this assert function into two to make it clearer and use the right one in the relevant tests...

i think at the moment this code might hide errors and make debugging difficult in the future.

---

@ndegwamartin can you elaborate further on how can we split the assertion here?

---

create a new function to assert two questionnaire responses in which you fiddle with the timestamps.

this assertion is used by a lot of tests. seems at least inefficient to do this operation for all resource types.

---

@jingtang10 @ndegwamartin  cc: @qiarie 

The updates I made to this method were not to test my own code, but to ensure the failing tests would pass. You can find the tests for the implementation [here](https://github.com/google/android-fhir/blob/98cb8662a67221abad7e80a2f73d79d6e11faaa8/datacapture/src/test/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaireResponsesTest.kt#L701).

After adding timestamps to the QuestionnaireResponses, many tests that asserted response equality started failing due to mismatching timestamps. Although the difference was only in milliseconds, it still caused the checks to fail. This left me with two options:

1. Update all the failing tests to use the same timestamp values for both launch and submission.
2. Modify the assertion method to programmatically adjust the expected and actual response timestamps.

I chose the second approach as it was more time-efficient, and we already have separate tests for timestamp functionality. I hope this makes sense.

---

The problem is that the function is now doing things that is quite specific to questionnaire responses and someone in the future is going to call the `assertResouceEquals` function and assume that it's just testing two resoures are equal but it's not... From a code health's perspective this is an anti pattern... we should make sure that the function does what the name suggests that it does.

Please create a function called `assertQuestionnaireResponseEqualsIgnoringLaunchTimestamp` which takes two `QuestionnaireResponse`s. In the test cases that you are seeing that fail - they should use this new funciton instead of the old function `assertResourceEquals`.

I think this change shouldn't be too big - it should be easy to find and replace usage of this function with questionnaire responses and use the new funciton.

---

also make sure you don't mutate the resources - make a copy and then remove the timestamp.